File: 0506-relative-ranks/0506-relative-ranks.js
/**
 * @param {number[]} score
 * @return {string[]}
 */
var findRelativeRanks = function(score) {
    const n = score.length;
    const pair = new Array(n);
    
    for (let i = 0; i < n; i++) {
        pair[i] = [score[i], i];
    }

    pair.sort((a, b) => b[0] - a[0]);
    const ans = new Array(n);

    for (let i = 0; i < n; i++) {
        if (i === 0) {
            ans[pair[i][1]] = "Gold Medal";
        } else if (i === 1) {
            ans[pair[i][1]] = "Silver Medal";
        } else if (i === 2) {
            ans[pair[i][1]] = "Bronze Medal";
        } else {
            ans[pair[i][1]] = String(i + 1);
        }
    }

    return ans;
};

File: 0506-relative-ranks/0506-relative-ranks.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
        int n = score.size();
        vector<vector<int>> curr(n, vector<int>(2));

        for(int i = 0; i<n; i++) {
            curr[i][0] = score[i];
            curr[i][1] = i;
        }

        sort(curr.begin(), curr.end(), [&](vector<int> &a, vector<int> &b){
            return a[0] > b[0];
        });
        vector<string> ans(n);

        for(int i = 0; i<n; i++) {
            if(i == 0) {
                ans[curr[i][1]] = "Gold Medal";
            }
            else if(i == 1) {
                ans[curr[i][1]] = "Silver Medal";
            }
            else if(i == 2) {
                ans[curr[i][1]] = "Bronze Medal";
            }
            else {
                ans[curr[i][1]] = to_string(i + 1);
            }
        }

        return ans;
    }
};